include/snmp_ip_count_table.h

```
@@ -0,0 +1,131 @@
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * The author can be reached by email at clearwater@metaswitch.com or by
+ * post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+ *
+ * Special Exception
+ * Metaswitch Networks Ltd  grants you permission to copy, modify,
+ * propagate, and distribute a work formed by combining OpenSSL with The
+ * Software, or a work derivative of such a combination, even if such
+ * copying, modification, propagation, or distribution would otherwise
+ * violate the terms of the GPL. You must comply with the GPL in all
+ * respects for all of the code used other than OpenSSL.
+ * "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+ * Project and licensed under the OpenSSL Licenses, or a work based on such
+ * software and licensed und er the OpenSSL Licenses.
+ * "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+ * under which the OpenSSL Project distributes the OpenSSL toolkit software,
+ * as those licenses appear in the file LICENSE-OPENSSL.
+ */
+
+#include <vector>
+#include <map>
+#include <string>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#include "snmp_table.h"
+#include "snmp_includes.h"
+#include "logger.h"
+
+#ifndef SNMP_IP_COUNT_TABLE_H
+#define SNMP_IP_COUNT_TABLE_H
+
+namespace SNMP
+{
+
+
+// Row of counters indexed by RFC 2851 IP addresses
+class IPCountRow : public Row
+{
+public:
+  IPCountRow(std::string addrStr) :
+    Row(),
+    _count(0)
+  {
+    // Convert the string into a type (IPv4 or IPv6) and a sequence of bytes
+    if (inet_pton(AF_INET, addrStr.c_str(), &_addr) == 1) {
+      _addr_type = 1; // IPv4
+      _addr_len = sizeof(struct in_addr);
+    } else if (inet_pton(AF_INET6, addrStr.c_str(), &_addr) == 1) {
+      _addr_type = 2; // IPv6
+      _addr_len = sizeof(struct in6_addr);
+    } else {
+      _addr_type = 0; // unknown
```

Is it right to construct a 0/0 OID in this case?  Is there not a chance we'll end up having two rows with the same index?  What happens then?  I think we should at least report an error.